include/boost/astronomy/time/parser.hpp

```diff
@@ -0,0 +1,66 @@
+/*=============================================================================
+Copyright 2020 Syed Ali Hasan <alihasan9922@gmail.com>
+
+Distributed under the Boost Software License, Version 1.0. (See accompanying
+file License.txt or copy at https://www.boost.org/LICENSE_1_0.txt)
+=============================================================================*/
+
+#ifndef BOOST_ASTRONOMY_PARSER_HPP
+#define BOOST_ASTRONOMY_PARSER_HPP
+
+#include <string>
+#include <iostream>
+
+using namespace std;
+
+struct decimal_hour {
+private:
+    double dh = 0;
+
+    double time = 0;
+    int hours = 0;
```

Comment:
with time we should use `long` instead of `int`

Reply:
There is a reason for using `int` as we are converting from decimal hours to hours, minutes, and seconds.

For example

42.756ℎ𝑟=42ℎ𝑟+0.756ℎ𝑟
0.756ℎ𝑟×60𝑚𝑖𝑛1ℎ𝑟=45.36𝑚𝑖𝑛
45.36𝑚𝑖𝑛=45𝑚𝑖𝑛+0.36𝑚𝑖𝑛
0.36𝑚𝑖𝑛×60𝑠1𝑚𝑖𝑛=21.6𝑠
21.6𝑠≈22𝑠

42:45:22 = 42 hours : 45 minutes : 22 seconds

Reply:
Updated parser.hpp
Returns hours and minutes as int and seconds as double.

`decimal_hour(42.756)` 

**returns** 

`42h 45m 21.600000s `